pytest_inmanta/plugin.py

```diff
@@ -763,13 +779,52 @@ def get_resource(
     """
 
     try:
-        resource = next(get_resources_matching(resources, resource_type, **filter_args))
+        resource = next(
+            get_resources_matching(
+                resources, resource_type, should_filter_model_type, **filter_args
+            )
+        )
         resource = check_serialization(resource)
         return resource
     except StopIteration:
         return None
 
 
+def get_one_resource(
+    resources: "collections.abc.Iterable[Resource]",
+    resource_type: str,
+    **filter_args: object,
+) -> typing.Optional[Resource]:
+    """
+    Get a resource of the given type and given filter on the resource attributes. This method makes sure that the
+    `entity_type` matches the provided `resource_type`, unlike the `get_resource` function
+    If multiple resource match, an assertion error is raised. If none match, None is returned.
+
+    :param resources: The resources to filter on
+    :param resource_type: The exact type used in the model (no super types)
+    """
+    resources = iter(
+        get_resources_matching(
+            resources,
+            resource_type,
+            should_filter_model_type=True,
+            **filter_args,
+        )
+    )
+
+    list_resources = []
+    for resource in resources:
+        resource = check_serialization(resource)
+        list_resources.append(resource)
+
+    # If we don't find anything matching these criteria, we should return `None`
+    if len(list_resources) == 0:
+        list_resources.append(None)
+
+    assert len(list_resources) == 1, "Only one resource should be found!"
```

Comment:
I would provide context to help debug this:

```suggestion
    assert len(list_resources) == 1, "The filter should only match one resource, but it matches: [{','.join(str(resource.id) for resource in list_resource)}]"
```